Userland/Libraries/LibGfx/ImageFormats/TGALoader.cpp

```diff
@@ -11,160 +11,34 @@
-        u8 green;
-        u8 red;
-        u8 alpha;
-    } components;
-
-    u32 data;
-};
-
-struct TGAPixelPacket {
-    bool raw;
-    u8 pixels_count;
-};
-
-static_assert(AssertSize<TGAPixel, 4>());
-
-class TGAReader {
-public:
-    TGAReader(ReadonlyBytes data)
-        : m_data(move(data))
-    {
-    }
-
-    TGAReader(ReadonlyBytes data, size_t index)
-        : m_data(move(data))
-        , m_index(index)
-    {
-    }
-
-    ALWAYS_INLINE u8 read_u8()
-    {
-        u8 value = m_data[m_index];
-        m_index++;
-        return value;
-    }
-
-    ALWAYS_INLINE i8 read_i8()
-    {
-        return static_cast<i8>(read_u8());
-    }
-
-    ALWAYS_INLINE u16 read_u16()
-    {
-        return read_u8() | read_u8() << 8;
-    }
-
-    ALWAYS_INLINE i16 read_i16()
-    {
-        return read_i8() | read_i8() << 8;
-    }
-
-    ALWAYS_INLINE u32 read_u32()
-    {
-        return read_u16() | read_u16() << 16;
-    }
-
-    ALWAYS_INLINE i32 read_i32()
-    {
-        return read_i16() | read_i16() << 16;
-    }
-
-    ALWAYS_INLINE TGAPixelPacket read_packet_type()
+struct TGAPixelData {
+    explicit TGAPixelData(ReadonlyBytes bytes)
     {
-        auto pixel_packet_type = read_u8();
-        auto pixel_packet = TGAPixelPacket();
-        pixel_packet.raw = !(pixel_packet_type & 0x80);
-        pixel_packet.pixels_count = (pixel_packet_type & 0x7f);
-
-        // NOTE: Run-length-encoded/Raw pixel packets cannot encode zero pixels,
-        // so value 0 stands for 1 pixel, 1 stands for 2, etc...
-        pixel_packet.pixels_count++;
-        return pixel_packet;
-    }
-
-    ALWAYS_INLINE TGAPixel read_pixel(u8 bits_per_pixel)
-    {
-        auto pixel = TGAPixel();
-
-        switch (bits_per_pixel) {
-        case 24:
-            pixel.components.blue = read_u8();
-            pixel.components.green = read_u8();
-            pixel.components.red = read_u8();
-            pixel.components.alpha = 0xFF;
-            return pixel;
-
-        case 32:
-            pixel.components.blue = read_u8();
-            pixel.components.green = read_u8();
-            pixel.components.red = read_u8();
-            pixel.components.alpha = read_u8();
-            return pixel;
-
-        default:
-            VERIFY_NOT_REACHED();
-        }
-    }
-
-    size_t index() const
-    {
-        return m_index;
-    }
-
-    ReadonlyBytes data() const
-    {
-        return m_data;
-    }
-
-private:
-    ReadonlyBytes m_data;
-    size_t m_index { 0 };
+        // NOTE: We support 24-bit color pixels and 32-bit color pixels
+        VERIFY(bytes.size() == 3 || bytes.size() == 4);
+        components.blue = bytes[0];
+        components.green = bytes[1];
+        components.red = bytes[2];
+        if (bytes.size() == 4)
+            components.alpha = bytes[3];
+        AK::TypedTransfer<u8>::copy(reinterpret_cast<u8*>(&raw), reinterpret_cast<u8*>(&components), sizeof(TGAColor));
+    }
+
+    struct [[gnu::packed]] TGAColor {
```

Comment:
It looks like this TGAColor/components is never used, so we can also get rid of that!
Finally, it seems that `TGAPixelData` is a `Gfx::Color` or at least a `ARGB32`, we should probably reuse one of them.